examples/geometry/src/components/circle.tsx

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable complexity */
+import {
+  forwardRef,
+  useContext,
+  useEffect,
+  useImperativeHandle,
+  useRef
+} from 'react';
+
+import type {Ref} from 'react';
+import {GoogleMapsContext, latLngEquals} from '@vis.gl/react-google-maps';
+
+type CircleEventProps = {
+  onClick?: (e: google.maps.MapMouseEvent) => void;
+  onDrag?: (e: google.maps.MapMouseEvent) => void;
+  onDragStart?: (e: google.maps.MapMouseEvent) => void;
+  onDragEnd?: (e: google.maps.MapMouseEvent) => void;
+  onMouseOver?: (e: google.maps.MapMouseEvent) => void;
+  onMouseOut?: (e: google.maps.MapMouseEvent) => void;
+  onRadiusChanged?: (r: ReturnType<google.maps.Circle['getRadius']>) => void;
+  onCenterChanged?: (p: ReturnType<google.maps.Circle['getCenter']>) => void;
+};
+
+export type CircleProps = google.maps.CircleOptions & CircleEventProps;
+
+export type CircleRef = Ref<google.maps.Circle | null>;
+
+function useCircle(props: CircleProps) {
+  const {
+    onClick,
+    onDrag,
+    onDragStart,
+    onDragEnd,
+    onMouseOver,
+    onMouseOut,
+    onRadiusChanged,
+    onCenterChanged,
+    radius,
+    center,
+    ...circleOptions
+  } = props;
+  // This is here to avoid triggering the useEffect below when the callbacks change (which happen if the user didn't memoize them)
+  const callbacks = useRef<Record<string, (e: unknown) => void>>({});
```

Comment:
That is indeed a nice way to handle this, I'll see if that pattern can be used for the map- and marker-events as well.

Reply:
😄